src/Raven.Server/Documents/Sharding/ShardedDocumentsStorage.cs

```diff
@@ -71,15 +69,48 @@ public ChangeVector GetMergedChangeVectorInBucket(DocumentsOperationContext cont
     {
         DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal,
             "Optimize this to calculate the merged change vector during insertion to the bucket");
-        DevelopmentHelper.ShardingToDo(DevelopmentHelper.TeamMember.Karmel, DevelopmentHelper.Severity.Normal,
-            "Extend this for all types not only docs");
 
         var table = new Table(DocsSchema, context.Transaction.InnerTransaction);
         var merged = context.GetChangeVector(string.Empty);
         foreach (var result in GetItemsByBucket(context.Allocator, table, DocsSchema.DynamicKeyIndexes[AllDocsBucketAndEtagSlice], bucket, 0))
         {
-            var document = TableValueToDocument(context, ref result.Result.Reader, DocumentFields.ChangeVector);
-            merged = merged.MergeWith(document.ChangeVector, context);
+            var documentCv = TableValueToChangeVector(context, (int)DocumentsTable.ChangeVector, ref result.Result.Reader);
+            merged = merged.MergeWith(documentCv, context);
+        }
+        foreach (var result in GetItemsByBucket(context.Allocator, table, TombstonesSchema.DynamicKeyIndexes[TombstonesBucketAndEtagSlice], bucket, 0))
```

Comment:
shouldn't taking last item yield the same result?

Reply:
yes you're right, fixed